sphinx_gallery/gen_gallery.py

```diff
@@ -461,41 +472,85 @@ def generate_gallery_rst(app):
 
     for examples_dir, gallery_dir in workdirs:
 
-        examples_dir = os.path.join(app.builder.srcdir, examples_dir)
-        gallery_dir = os.path.join(app.builder.srcdir, gallery_dir)
+        examples_dir_abs_path = os.path.join(app.builder.srcdir, examples_dir)
+        gallery_dir_abs_path = os.path.join(app.builder.srcdir, gallery_dir)
 
-        # Here we don't use an os.walk, but we recurse only twice: flat is
-        # better than nested.
-        this_fhindex, this_costs = generate_dir_rst(
-            examples_dir, gallery_dir, gallery_conf, seen_backrefs)
+        # Create section rst files and fetch content which will
+        # be added to current index file
+        this_fhindex, this_costs, this_toctree, _ = generate_dir_rst(
+            examples_dir_abs_path,
+            gallery_dir_abs_path,
+            gallery_conf,
+            seen_backrefs
+        )
+
+        # Filter out tags from fhindex
+        # to prevent tag duplication across the documentation
+        tag_regex = r"^\.\.(\s+)\_(.+)\:(\s*)$"
+        this_fhindex = "\n".join(list(
+            filter(
+                lambda l: re.match(tag_regex, l) is None,
+                iter(this_fhindex.splitlines())
+            )
+        ))
```

Comment:
You don't need to construct a `list` as `join` can operate on an iterable. Also I think `filter` and `iter` here end up making it more complicated than the direct alternative below, WDYT?
```suggestion
        this_fhindex = "\n".join(
            line for line in this_fhindex.splitlines()
            if re.match(tag_regex, line) is None
        )
```